plugins/module_utils/foreman_helper.py

```diff
@@ -357,20 +358,31 @@ def _delete_entity(self, resource, current_entity, params):
         payload = {'id': current_entity['id']}
         if params:
             payload.update(params)
-        self._resource_action(resource, 'destroy', payload)
+        self.resource_action(resource, 'destroy', payload)
         return True, None
 
-    def _resource_action(self, resource, action, params):
+    def resource_action(self, resource, action, params, **kwargs):
         resource_payload = self.foremanapi.resource(resource).action(action).prepare_params(params)
         try:
             result = None
             if not self.check_mode:
-                result = self.foremanapi.resource(resource).call(action, resource_payload)
+                result = self.foremanapi.resource(resource).call(action, resource_payload, **kwargs)
         except Exception as e:
             self.fail_json(msg='Error while performing {} on {}: {}'.format(
                 action, resource, str(e)))
         return True, result
 
+    def wait_for_task(self, task, duration=60, poll=4):
+        if task['state'] not in ['paused', 'stopped']:
+            if duration > 0:
+                time.sleep(poll)
+                duration -= poll
+                _, task = self.resource_action('foreman_tasks', 'show', {'id': task['id']})
+                task = self.wait_for_task(task, duration, poll)
```

Comment:
Could this use a while loop instead of recursion?

Reply:
I heard of a theorem...

Reply:
see https://github.com/theforeman/foreman-ansible-modules/pull/392/files#r311952665? :)